src/main/scala/scorex/core/consensus/ModifierSemanticValidity.scala

```diff
@@ -1,7 +1,30 @@
 package scorex.core.consensus
 
-sealed trait ModifierSemanticValidity
-case object Absent extends ModifierSemanticValidity
-case object Unknown extends ModifierSemanticValidity
-case object Valid extends ModifierSemanticValidity
-case object Invalid extends ModifierSemanticValidity
\ No newline at end of file
+sealed trait ModifierSemanticValidity {
+  val code: Byte
+}
+
+object ModifierSemanticValidity {
+  def restoreFromCode(code: Byte): ModifierSemanticValidity = code match {
+    case b: Byte if b == Absent.code => Absent
```

Comment:
`if ... else if ... else` would be more succinct and cleaner than pattern matching here.

Reply:
And more efficient too.